src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -74,11 +83,28 @@ trait ClusterComponent extends LeoComponent {
       }
     }
 
-    def deleteCluster(googleId: UUID):DBIO[Int] = {
-       clusterQuery.filter(_.googleId === googleId)
-         .map(c => (c.destroyedDate, c.status))
-         .update((Option(Timestamp.from(java.time.Instant.now())), ClusterStatus.Deleting.toString))
+    def markPendingDeletion(googleId: UUID): DBIO[Int] = {
+      clusterQuery.filter(_.googleId === googleId)
+        .map(c => (c.destroyedDate, c.status))
+        .update((Option(Timestamp.from(java.time.Instant.now())), ClusterStatus.Deleting.toString))
+    }
+
+    def completeDeletion(googleId: UUID, clusterName: String): DBIO[Int] = {
+      updateClusterStatus(googleId, ClusterStatus.Deleted) andThen
+        // Append a random suffix to the cluster name to prevent unique key conflicts in case a cluster
+        // with the same name is recreated.
+        // TODO: This is a bit ugly; a better solution would be to have a unique key on (googleId, clusterName, destroyedDate)
```

Comment:
Discussion on this is welcome; hopefully this is an OK interim solution.

Reply:
figuring this out for realz is on my list of leo housekeeping tasks for post-demo

Reply:
👍

Reply:
We do the same thing in Rawls (EntityComponent and MethodCongurationComponent)

https://broadinstitute.atlassian.net/browse/GAWB-1796

Reply:
Yeah it's where I drew my inspiration :)

Reply:
Although it looks like Rawls takes better care to avoid collisions. Probably not worth it here if we're planning on revisiting anyway.